cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/FlowletProgramRunner.java

```diff
@@ -541,8 +542,10 @@ private ProcessSpecificationFactory processSpecificationFactory(
               && (inputNames.contains(queueName.getSimpleName())
               || inputNames.contains(FlowletDefinition.ANY_INPUT))) {
 
-              if (entry.getKey().getType() == FlowletConnection.Type.STREAM) {
-                ConsumerSupplier<StreamConsumer> consumerSupplier = ConsumerSupplier.create(program.getNamespace(),
+              Node sourceNode = entry.getKey();
+              if (sourceNode.getType() == FlowletConnection.Type.STREAM) {
+                 NamespaceId sourceNamespace = new NamespaceId(sourceNode.getNamespace());
```

Comment:
Yes. But when the type of a sourcenode is STREAM, the code here https://github.com/caskdata/cdap/pull/6003/files#diff-6533ab2131f2a4fbc1e718ac2e27cafaR87 will guarantee that the getNamespace won't return null and it passed the test cases. 

Since a node can either be a STREAM type or other type(where getNamespace will always return null), putting things together make this actually crappy.